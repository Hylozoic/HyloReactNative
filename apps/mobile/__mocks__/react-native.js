import React, {useState} from 'react'
import * as ReactNative from 'react-native'

export const Switch = props => {
  const [value, setValue] = useState(props.value)
  return (
    <ReactNative.TouchableOpacity
      onPress={() => {
        props.onValueChange(!value)
        setValue(!value)
      }}
      testID={props.testID}>
      <ReactNative.Text>{value?.toString()}</ReactNative.Text>
    </ReactNative.TouchableOpacity>
  )
}

newRN = Object.defineProperty(ReactNative, 'Switch', {
  get: function() {
    return Switch
  }
})

module.exports = newRN
