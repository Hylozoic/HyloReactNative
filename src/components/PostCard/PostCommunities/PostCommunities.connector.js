import { connect } from 'react-redux'
import { filter, get } from 'lodash/fp'

export function mapStateToProps (state, props) {
  const communities = props.selectedNetworkId
    ? filter(community => get('network.id', community) === props.selectedNetworkId, props.communities)
    : props.communities

  return {
    communities
  }
}

export default connect(mapStateToProps)
