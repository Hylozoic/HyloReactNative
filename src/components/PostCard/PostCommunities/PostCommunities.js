import React, { Component } from 'react'
import { get, isEmpty, chunk } from 'lodash/fp'
import Icon from '../../Icon'
import { Text, View, TouchableOpacity, Image } from 'react-native'
import { DEFAULT_AVATAR } from '../../../store/models/Community'
import {
  capeCod10, rhino30, caribbeanGreen
} from '../../../style/colors'

export default class PostCommunities extends Component {
  static defaultState = {
    expanded: false
  }

  constructor (props) {
    super(props)
    this.state = PostCommunities.defaultState
  }

  toggleExpanded = () => {
    this.setState({
      expanded: !this.state.expanded
    })
  }

  render () {
    const { communities, slug, goToCommunity } = this.props

    // don't show if there are no communities or this isn't cross posted

    if (isEmpty(communities) || (communities.length === 1 && get('0.slug', communities) === slug)) return null

    const { expanded } = this.state

    const content = expanded
      ? <View style={styles.expandedSection}>
        <View style={styles.row}>
          <Text style={styles.reminderText}>Posted In: </Text>
          <TouchableOpacity onPress={this.toggleExpanded} style={styles.arrowButton}>
            <Icon name='ArrowUp' style={styles.arrowIcon} />
          </TouchableOpacity>
        </View>
        {chunk(2, communities).map(pair => <CommunityRow communities={pair} key={pair[0].id} goToCommunity={goToCommunity} />)}
      </View>
      : <View style={styles.row}>
        <Text style={styles.reminderText}>Posted In: </Text>
        <CommunityList communities={communities} expandFunc={this.toggleExpanded} goToCommunity={goToCommunity} />
        <TouchableOpacity onPress={this.toggleExpanded} style={styles.arrowButton}>
          <Icon name='ArrowDown' style={styles.arrowIcon} />
        </TouchableOpacity>
      </View>

    return <View style={[styles.communities, expanded && styles.expanded]}>
      {content}
    </View>
  }
}

export function CommunityList ({communities, goToCommunity, expandFunc}) {
  const moreCommunities = communities.length > 1
  const othersText = n => n === 1 ? '1 other' : `${n} others`
  return <View style={[styles.communityList, styles.row]}>
    <TouchableOpacity onPress={() => goToCommunity(communities[0].id)} style={{flex: -1}}><Text style={styles.linkText} numberOfLines={1}>{communities[0].name}</Text></TouchableOpacity>
    {moreCommunities && <View style={[styles.row, {flex: 0}]}>
      <Text style={[styles.reminderText]}> and </Text>
      <TouchableOpacity onPress={expandFunc}><Text style={styles.linkText}>{othersText(communities.length - 1)}</Text></TouchableOpacity>
    </View>}
  </View>
}

export function CommunityRow ({ communities, goToCommunity }) {
  return <View style={[styles.communityRow, styles.row]}>
    {communities.map(community => <CommunityCell key={community.id} community={community} goToCommunity={goToCommunity} />)}
  </View>
}

export function CommunityCell ({ community, goToCommunity }) {
  const { name, avatarUrl } = community
  const imageSource = {uri: avatarUrl || DEFAULT_AVATAR}

  return <TouchableOpacity style={[styles.communityCell, styles.row]} onPress={() => goToCommunity(community.id)}>
    <Image source={imageSource} style={styles.communityAvatar} />}
    <Text style={[styles.linkText, styles.communityCell]} numberOfLines={1}>{name}</Text>
  </TouchableOpacity>
}

const styles = {
  communities: {
    borderTopWidth: 1,
    borderBottomWidth: 1,
    borderColor: capeCod10,
    paddingHorizontal: 12
  },
  row: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center'
  },
  communityList: {
    height: 40,
    justifyContent: 'flex-start'
  },
  arrowButton: {
    marginLeft: 'auto'
  },
  arrowIcon: {
    color: rhino30,
    fontSize: 18
  },
  reminderText: {
    color: rhino30,
    fontSize: 13,
    fontFamily: 'Circular-Book'
  },
  linkText: {
    color: caribbeanGreen,
    fontSize: 13,
    fontFamily: 'Circular-Book'
  },
  expandedSection: {
    paddingTop: 11,
    paddingBottom: 5
  },
  communityRow: {
    paddingVertical: 8
  },
  communityCell: {
    paddingRight: 15
  },
  communityAvatar: {
    height: 20,
    width: 20,
    borderRadius: 4,
    marginRight: 9
  }
}
