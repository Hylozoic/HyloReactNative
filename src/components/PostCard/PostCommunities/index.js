import PostCommunities from './PostCommunities'

export default PostCommunities
