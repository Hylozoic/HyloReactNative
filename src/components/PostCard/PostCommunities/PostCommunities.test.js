import 'react-native'
import React from 'react'
import ReactTestRenderer from 'react-test-renderer'
import PostCommunities from './PostCommunities'

describe('PostCommunities', () => {
  it('matches last snapshot', () => {
    const props = {
      communities: [
        {
          id: 1,
          name: 'One',
          slug: 'one'
        },
        {
          id: 2,
          name: 'Two',
          slug: 'two'
        },
        {
          id: 3,
          name: 'Three',
          slug: 'three'
        }
      ],
      slug: 'hylo'
    }

    const renderer = ReactTestRenderer.create(<PostCommunities {...props} />)
    expect(renderer).toMatchSnapshot()
  })

  it('matches last snapshot when expanded', () => {
    const props = {
      communities: [
        {
          id: 1,
          name: 'One',
          slug: 'one',
          avatarUrl: 'foo.png'
        },
        {
          id: 2,
          name: 'Two',
          slug: 'two'
        },
        {
          id: 3,
          name: 'Three',
          slug: 'three'
        }
      ],
      slug: 'hylo'
    }

    const renderer = ReactTestRenderer.create(<PostCommunities {...props} />)
    renderer.getInstance().toggleExpanded()
    expect(renderer).toMatchSnapshot()
  })

  it('returns null when in the only community', () => {
    const props = {
      communities: [
        {
          id: 1,
          name: 'One',
          slug: 'one'
        }
      ],
      slug: 'one'
    }

    const renderer = ReactTestRenderer.create(<PostCommunities {...props} />)
    expect(renderer).toMatchSnapshot()
  })
})
