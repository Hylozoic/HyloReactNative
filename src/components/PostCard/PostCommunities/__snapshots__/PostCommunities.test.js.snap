// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCommunities matches last snapshot 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Posted In: 
    </Text>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
        >
          One
        </Text>
      </View>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
           and 
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            2 others
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Icon
        color={undefined}
        name="ArrowDown"
      />
    </View>
  </View>
</View>
`;

exports[`PostCommunities matches last snapshot when expanded 1`] = `
<View>
  <View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Posted In: 
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Icon
          color={undefined}
          name="ArrowUp"
        />
      </View>
    </View>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={
            Object {
              "uri": "foo.png",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
        >
          One
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={
            Object {
              "uri": "https://d3ngex8q79bk55.cloudfront.net/misc/default_community_avatar.png",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
        >
          Two
        </Text>
      </View>
    </View>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={
            Object {
              "uri": "https://d3ngex8q79bk55.cloudfront.net/misc/default_community_avatar.png",
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
        >
          Three
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`PostCommunities renders when in one community and shouldShowCommunities is true 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Posted In: 
    </Text>
    <View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
        >
          One
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Icon
        color={undefined}
        name="ArrowDown"
      />
    </View>
  </View>
</View>
`;

exports[`PostCommunities returns null when in one community and shouldShowCommunities is false 1`] = `null`;
